### DISCUSSION TITLE
Combobox virtualization

### DISCUSSION THREAD

**Discussion by PabloFNK** [twilio-labs/paste#2301] [Q&A] [answered]
Hi team,

I am working on implementing something like the [Filter Group pattern](https://paste.twilio.design/patterns/filter-group/). The thing is that I have a date picker with a combobox as suggested with several options and the last one is the "Custom" option that opens the date picker popover.

The thing is that I would like to test this behaviour but as the Combobox virtualizes it only to the 5 first options it is not available to click on the DOM.

Do you recommend some way to test this? Mock the virtualization library? Force an scroll action on the list to see if that shows it in the DOM?

**Comment by gloriliale** [comment]
Hi @PabloFNK! Consulting the team about this now, thanks for the question!

**Comment by nkrantz** [comment, marked as answer]
Hey @PabloFNK - we'd recommend mocking your data (if possible), and testing the component functionality independently. If you're writing E2E tests and you're unable to limit the dataset, you will likely need to scroll the list to find the option you want to click on.
